src/open_clip/tokenizer.py

```diff
@@ -187,10 +187,18 @@ def tokenize(texts: Union[str, List[str]], context_length: int = 77) -> torch.Lo
 
 class HFTokenizer:
     "HuggingFace tokenizer wrapper"
-    def __init__(self, tokenizer_name:str):
+    def __init__(self, tokenizer_name:str, add_pad_and_mask_tokens=False):
         from transformers import AutoTokenizer
         self.tokenizer = AutoTokenizer.from_pretrained(tokenizer_name)
 
+        if add_pad_and_mask_tokens:
+            if self.tokenizer.pad_token_id is None:
+                assert '<|pad|>' not in self.tokenizer.get_vocab()
+                self.tokenizer.add_special_tokens({'pad_token': '<|pad|>'})
+            if self.tokenizer.mask_token_id is None:
+                assert '<|mask|>' not in self.tokenizer.get_vocab()
+                self.tokenizer.add_special_tokens({'mask_token': '<|mask|>'})
+
```

Comment:
Original FLAVA doesn't use anything special just BERT tokenizer. Would it make sense to add original FLAVA version first and then move on to customizations?

Reply:
Instead of doing it this way, I would suggest using HuggingFace hub to save the tokenizer with custom tokens and then loading the custom tokenizer directly. Then this custom code logic doesn't need to be added here which is anyways very specifically to experiment we will run.